src/Symfony/Component/VarExporter/Hydrator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarExporter;
+
+use Symfony\Component\VarExporter\Internal\Hydrator as InternalHydrator;
+
+/**
+ * Utility class to hydrate the properties of an object.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class Hydrator
+{
+    /**
+     * Sets the properties of an object, including private and protected ones.
+     *
+     * For example:
+     *
+     *     // Sets the public or protected $object->propertyName property
+     *     Hydrator::hydrate($object, ['propertyName' => $propertyValue]);
+     *
+     *     // Sets a private property defined on its parent Bar class:
+     *     Hydrator::hydrate($object, ["\0Bar\0privateBarProperty" => $propertyValue]);
+     *
+     *     // Alternative way to set the private $object->privateBarProperty property
+     *     Hydrator::hydrate($object, [], [
+     *         Bar::class => ['privateBarProperty' => $propertyValue],
+     *     ]);
+     *
+     * Instances of ArrayObject, ArrayIterator and SplObjectHash can be hydrated
```

Comment:
I think you mean `SplObjectStorage`, not `SplObjectHash` (which does not exist)

Reply:
fixed, thanks